FirebaseSwiftUI/FirebaseFacebookSwiftUI/Sources/Views/FacebookButtonView.swift

```diff
@@ -0,0 +1,108 @@
+import AppTrackingTransparency
+import FacebookCore
+import FacebookLogin
+import FirebaseAuth
+import FirebaseAuthSwiftUI
+import SwiftUI
+
+@MainActor
+public struct FacebookButtonView {
+  @Environment(AuthService.self) private var authService
+  @State private var errorMessage = ""
+  @State private var showCanceledAlert = false
+  @State private var showUserTrackingAlert = false
+  @State private var limitedLogin = true
+
+  public init() {}
+
+  private var limitedLoginBinding: Binding<Bool> {
+    Binding(
+      get: { self.limitedLogin },
+      set: { newValue in
+        let trackingStatus = ATTrackingManager.trackingAuthorizationStatus
+        if newValue == true, trackingStatus != .authorized {
+          self.showUserTrackingAlert = true
```

Comment:
Instead of changing this from inside a computed property, I would recommend using the `onChange(of:)` view modifier to listen to `limitedLogin` and update `showUserTrackingAlert` accordingly.